### DISCUSSION TITLE
New Source? Or way of configuring manually?

### DISCUSSION THREAD

**Discussion by HeadJHA** [mampfes/hacs_waste_collection_schedule#1287] [Q&A]
Hey, New to HA, and i see config from UI isnt quite suitable for this integration - so i will be taking my first foray into .yaml with this integration i guess!

I would like to get the Mid-Suffolk District (United Kingdom) source into this..  https://www.midsuffolk.gov.uk/waste-services/collection-days/

They seem to have a pdf for which bin is on which week
https://www.midsuffolk.gov.uk/assets/Waste-Services/Calendars/MID-SUFFOLK-Refuse-Recycling-Collection-Schedule-2023.pdf

Then there is a tool for checking which day - where you input a first line address rather than a postcode... haven't been able to find an iCal etc.

Any ideas from you guys more in the know as to the best way I can do this?

Many thanks

**Comment by 5ila5** [comment]
PDF parsing does not work well and we generally avoided doing this. 
The best way is to write a source which parses the html. If you don't want or can do it yourself feel free to open an Issue.
Do you know what the number in the url (https://bmsdchosting.net/waste-services/mid-suffolk-bin-collection/weeks/12132) represents? it may just be an internal id.

If you want to try it yourself here is the [contribution guide for dedicated sources](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md)

**Reply by HeadJHA** [reply]
I’m probably not capable of doing this properly. I don’t know what the URL ending means no
